tests/run-make/short-ice/rmake.rs

```diff
@@ -0,0 +1,41 @@
+// Backtraces in internal compiler errors used to be unbearably long, spanning
+// multiple hundreds of lines. A fix was pushed in #108938, and this test gathers
+// varied metrics on level 1 and full-level backtraces to check that the output
+// was shortened down to an appropriate length.
+// See https://github.com/rust-lang/rust/issues/107910
+
+use run_make_support::rustc;
+use std::env;
+
+fn main() {
+    env::set_var("RUST_BACKTRACE", "1");
+    let mut rust_test_1 = rustc().input("src/lib.rs").arg("-Ztreat-err-as-bug=1").run_fail();
+    env::set_var("RUST_BACKTRACE", "full");
+    let mut rust_test_2 = rustc().input("src/lib.rs").arg("-Ztreat-err-as-bug=1").run_fail();
+    let mut log_1 = rust_test_1.stdout;
+    log_1.append(&mut rust_test_1.stderr);
+    let mut log_2 = rust_test_2.stdout;
+    log_2.append(&mut rust_test_2.stderr);
+    let rust_test_log_1 = &std::str::from_utf8(&log_1).unwrap();
+    let rust_test_log_2 = &std::str::from_utf8(&log_2).unwrap();
+
+    let rustc_query_count_full = count_lines_with(rust_test_log_2, "rustc_query_");
+
+    assert!(
+        rust_test_log_1.lines().count() < rust_test_log_2.lines().count()
+            && count_lines_with(rust_test_log_2, "__rust_begin_short_backtrace")
+                == count_lines_with(rust_test_log_2, "__rust_end_short_backtrace")
+            && count_lines_with(rust_test_log_1, "rustc_query_") + 5 < rustc_query_count_full
+            && rustc_query_count_full > 5
+    );
+}
+
+fn count_lines_with(s: &str, search: &str) -> usize {
```

Comment:
We don't need to do it here, but this could be rewritten as
```rust
fn count2(s: &str, search: &str) -> usize {
    s.lines().filter(|l| l.contains(search)).count()
}
```

Reply:
So elegant! Added.